mbed-client/m2mbase.h

```diff
@@ -243,13 +243,13 @@ class M2MBase : public M2MReportObserver {
      * \brief Returns the interface description of the object.
      * \return Description of the object.
      */
-    virtual const String& interface_description() const;
+    String interface_description() const;
 
     /**
      * \brief Returns the resource type of the object.
      * \return Resource type of the object.
      */
-    virtual const String& resource_type() const;
+     String resource_type() const;
```

Comment:
This breaks API and the new behaviour (alloc a copy of data) needs to be documented on comments.